app/session.server.ts

```diff
@@ -0,0 +1,54 @@
+import { createCookieSessionStorage, Session } from "@remix-run/node";
+import { auth } from "./firebase.server";
+
+const sessionCookieSecret = process.env.SESSION_COOKIE_SECRET;
+if (!sessionCookieSecret) {
+  throw new Error(
+    "Please provide a SESSION_COOKIE_SECRET environment variable"
+  );
+}
+
+const { getSession, commitSession, destroySession } =
+  // Remix's built-in session handling
+  // https://remix.run/docs/en/v1/api/remix#sessions
+  createCookieSessionStorage({
```

Comment:
Remix provides a wrapper for session cookies out of the box, it's really handy. It requires a `sessionCookieSecret` for hashing purposes, but it's completely fine to fake it locally because we don't require production credentials to log in.